### ISSUE TITLE
7$: Implement TypeScript's stricter redeclaration rules

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#757]
The following is valid JS, but invalid TS:
```js
var x;
function x() {}
```
Let's report a redeclaration error when in TypeScript mode.
